casper/casper-spec.ini

```diff
@@ -44,6 +44,128 @@ requires:
++requires:
+    andBool TOTAL_CURDYN_DEPOSITS_VAL =/=Int 0
+    andBool TOTAL_PREVDYN_DEPOSITS_VAL =/=Int 0
+    andBool RESULT_1 ==Int (CUR_DYN_VOTES_VAL *Int VYPER_DEC_MULTIPLIER /Int TOTAL_CURDYN_DEPOSITS_VAL)
+    andBool RESULT_2 ==Int (PREV_DYN_VOTES_VAL *Int VYPER_DEC_MULTIPLIER /Int TOTAL_PREVDYN_DEPOSITS_VAL)
+    andBool RESULT_1 <=Int RESULT_2
+ensures:
+    ensures 0 <=Int RESULT_1 andBool RESULT_1 <=Int VYPER_DEC_MULTIPLIER
+
+[main_hash_voted_frac-success-2]
+k: (#execute => #halt) ~> _
+statusCode: _ => EVMC_SUCCESS
+output: _ => #asByteStackInWidth(RESULT_2, 32)
++requires:
+    andBool TOTAL_CURDYN_DEPOSITS_VAL =/=Int 0
+    andBool TOTAL_PREVDYN_DEPOSITS_VAL =/=Int 0
+    andBool RESULT_1 ==Int (CUR_DYN_VOTES_VAL *Int VYPER_DEC_MULTIPLIER /Int TOTAL_CURDYN_DEPOSITS_VAL)
+    andBool RESULT_2 ==Int (PREV_DYN_VOTES_VAL *Int VYPER_DEC_MULTIPLIER /Int TOTAL_PREVDYN_DEPOSITS_VAL)
+    andBool RESULT_1 >Int RESULT_2
+ensures:
+    ensures 0 <=Int RESULT_2 andBool RESULT_2 <=Int VYPER_DEC_MULTIPLIER
+
+[main_hash_voted_frac-failure]
+k: (#execute => #halt) ~> _
+statusCode: _ => EVMC_REVERT
++requires:
+    andBool (TOTAL_CURDYN_DEPOSITS_VAL ==Int 0 orBool TOTAL_PREVDYN_DEPOSITS_VAL ==Int 0)
+
+;def total_curdyn_deposits_scaled() -> wei_value:
+;    return floor(self.total_curdyn_deposits * self.deposit_scale_factor[self.current_epoch])
+[total_curdyn_deposits_scaled]
+comment:
+k: (#execute => #halt) ~> _
+output: _ => #asByteStackInWidth(RESULT, 32)
+statusCode: _ => EVMC_SUCCESS
+callStack: _
+callData: #abiCallData("total_curdyn_deposits_scaled", .TypedArgs)
+callValue: 0
+wordStack: .WordStack => _
+localMem: .Map => _
+pc: 0 => _
+gas: 100000 => _
+memoryUsed: 0 => _
+callDepth: CD
+log: _
+refund: _
+coinbase: _
+activeAccounts: _
+casperBalance: _
+storage:
+    // self.total_curdyn_deposits : decimal
+    ((#hashedLocation({COMPILER}, {TOTAL_CURDYN_DEPOSITS}, .IntList)  |-> TOTAL_CURDYN_DEPOSITS))
+    // self.current_epoch : int128
+    (#hashedLocation({COMPILER}, {CURRENT_EPOCH}, .IntList) |-> CURRENT_EPOCH)
+    //self.deposit_scale_factor[self.current_epoch] : decimal
+    (#hashedLocation({COMPILER}, {DEPOSIT_SCALE_FACTOR}, CURRENT_EPOCH) |-> CUR_EPOCH_SCALE_FACTOR)
+    _:Map
+accounts:
+requires:
+    andBool 0 <=Int TOTAL_CURDYN_DEPOSITS   andBool TOTAL_CURDYN_DEPOSITS   <=Int maxDecimal
+    andBool 0 <=Int CURRENT_EPOCH           andBool CURRENT_EPOCH           <=Int maxSInt128
+    andBool 0  <Int CUR_EPOCH_SCALE_FACTOR  andBool CUR_EPOCH_SCALE_FACTOR  <=Int maxDecimal
+    andBool TOTAL_CURDYN_DEPOSITS *Int CUR_EPOCH_SCALE_FACTOR               <Int pow255
+    andBool DECIMAL_RESULT ==Int TOTAL_CURDYN_DEPOSITS *Int CUR_EPOCH_SCALE_FACTOR /Int VYPER_DEC_MULTIPLIER
+    andBool DECIMAL_RESULT <=Int maxDecimal
+    andBool RESULT ==Int DECIMAL_RESULT /Int VYPER_DEC_MULTIPLIER
+ensures:
+    ensures 0 <=Int RESULT                  andBool RESULT <=Int maxSInt128
+attribute:
+
```

Comment:
@denis-bogdanas how about the negative cases for `total_curdyn_deposits_scaled`?